Makefile

```diff
@@ -884,8 +884,11 @@ install.tools: .install.goimports .install.gitvalidation .install.md2man .instal
 .install.golangci-lint: .gopathok
 	VERSION=1.45.2 GOBIN=$(GOBIN) ./hack/install_golangci.sh
 
+# N/B: An executable 'hack/bats' exists, make sure it's not
+# found by the installer script.
 .PHONY: .install.bats
 .install.bats: .gopathok
+	$(eval unexport PATH)
```

Comment:
I'm sorry, I can't allow this. It's incredibly dangerous: I've just spent a long time testing it, and have confirmed that the `unexport` carries over for the lifetime of the `make`. So this is fine for someone calling `make .install.bats`, or (shudder) in the current case where `.install.bats` is the last component of `install.tools`, but should anyone ever add anything to `install.tools` it will fail catastrophically with a difficult-to-debug error.

The correct solution, of course, is that `hack` should not be in anyone's path ever.

Reply:
> carries over for the lifetime of the make

Ugh, god, thanks for catching that.  Yeah, I really struggled with this fix and even trying to understand how/why it suddenly became a problem.  I tracked down the path modification to a change (facc009ca0) @vrothberg made at the end of April.  It appears related to a benchmarking change that I can't understand.

Valentin: The path change is causing a problem where `hack/install_bats.sh` is incorrectly finding `hack/bats`, so therefore failing to install when it's really needed.  Is there a simple change we can make so your benchmarks don't need `hack` in their path (since it's dangerous, as Ed pointed out).

Reply:
While waiting for @vrothberg, I just force-pushed a commit that fixes the `Makefile`.  Lets see what breaks, if anything...

Reply:
@cevich, feel free to revert this specific change.  I can export the path in `make localbenchmarks`.  I added it to have some scripts in ./hack in the PATH.

Reply:
As Ed found out, probably the only safe way to do it in a recipe (that doesn't affect every other target) would be something like:

```Makefile
.PHONY: localbenchmarks
localbenchmarks:
    env PATH=$(PATH):$(CURDIR)/hack command-to-run
```

Just be aware, this can cause some really strange and hard to debug side-effects, like `command-to-run` picking up `hack/bats` instead of `/usr/bin/bats` (for example) :wink: